build-system/server/app-utils.js

```diff
@@ -61,6 +63,8 @@ function logServeMode() {
   const serveMode = getServeMode();
   if (serveMode == 'compiled') {
     log(green('Serving'), cyan('minified'), green('JS'));
+  } else if (serveMode == 'esm') {
```

Comment:
since esm would ideally be the "default", would it makes sense for us to make it so? (not saying everything should be changed in this PR, just long term)

Reply:
Today, unminified is the default, and I think we should keep it so for local development. For minified, we'll now have two modes: `--compiled` and `--esm`. Changing the default server behavior to include `esm` can be done trivially by always injecting the `.mjs` alternative, but will need changes to how code is built, and how tests are run.

We can certainly consider this for the long term.

Reply:
Considering how quickly we are moving to the default output of the Google AMP Cache to module/nomodule... it's the default very quickly.

Perhaps we shouldn't address in this PR, but the local development environment should mirror our best intentions with the Cache.

Reply:
There are several advantages to using unminified code for local development that include the ability to doing lazy builds, faster transforms that don't use closure compiler, and the ability to do measure things like code coverage.

Therefore, I don't think the default local development mode should switch to minified builds. However, I do agree that esm should become the default when minified builds are used during local development. (Does this distinction make sense?)

One way to make esm the default for minified builds is to change `gulp dist` to generate both `.js` and `.mjs`, and modify our CI to run tests for both modes. Can investigate and work on this outside the context of this PR.

Reply:
Perhaps there is a missing item here in the discussion.

For unminifed builds, the suggestion is to generate both an ESM and nomodule output for testing.

Reply:
Gotcha. I’m sure it’s possible. Let’s discuss via a new FR.

Reply:
I've filed #27386 to separately track this effort.